### PR TITLE
Add device virtual removal on error before exit

### PR DESCRIPTION
**Pull request by farshidtz**
Also, add manual connection in case a local build of the platform snap is given.

### CODE REVIEW

File: data/latest/test-rules-engine.sh [hunk 1]
```diff
@@ -21,6 +21,12 @@ START_TIME=$(date +"%Y-%m-%d %H:%M:%S")
 
 snap_remove
 
+snap_remove_all()
+{
+    snap_remove
+    snap remove edgex-device-virtual
+}
+
```
Comment (MonicaisHer) [inline data/latest/test-rules-engine.sh]:
```suggestion

snap_remove_all()
{
    snap_remove
    snap remove edgex-device-virtual
}
snap_remove_all
```

**Review by MonicaisHer — CHANGES_REQUESTED** [review]
line 150 and 198 also need to be updated to `snap_remove_all()`

**Comment by farshidtz** [comment]
> line 150 and 198 also need to be updated to `snap_remove_all()`

Those have been removed from main by #47. Changing here will cause conflicts.

**Review by MonicaisHer — APPROVED** [review]
@farshidtz Thank you, looks good.

**Pull request merged**